File: TEMP.ViewModels/InsertRibbonTabViewModel.fs
namespace TEMP.ViewModels

open System
open System.ComponentModel
open SafetyProgram.UI.Views.MainView.Default
open SafetyProgram.Core
open Fluent
open System.Collections.ObjectModel
open SafetyProgram.Core.Commands.ICommands
open SafetyProgram.Core.Models
open SafetyProgram.UI.Views.ModelViews.RibbonTabViews.Documents

type InsertRibbonTabViewModel(model, commandInvoker) = 
    let commands = new DocumentICommands(
                    model,
                    commandInvoker)
    
    interface IInsertRibbonTabViewModel with
        member this.Commands = commands :> IDocumentICommands

File: SafetyProgram.UI.Views/ModelViews/RibbonTabViews/Documents/InsertTab/InsertRibbonTabViewModel.cs
using SafetyProgram.Base;
using SafetyProgram.Core.Commands.ICommands;
using SafetyProgram.Core.Models;

namespace SafetyProgram.UI.Views.ModelViews.RibbonTabViews.Documents
{
    /// <summary>
    /// Defines a standard implementation of an IInsertRibbonTabViewModel.
    /// </summary>
    public sealed class InsertRibbonTabViewModel : IInsertRibbonTabViewModel
    {
        /// <summary>
        /// Construct an instance of a viewmodel for an insert ribbon.
        /// </summary>
        /// <param name="commands">A set of commands that act on the document that items will be inserted into.</param>
        public InsertRibbonTabViewModel(IDocument model, 
            ICommandInvoker commandInvoker)
        {
            Helpers.NullCheck(model, commandInvoker);

            this.Commands = new DocumentICommands(
                model,
                commandInvoker);
        }

        /// <summary>
        /// Get a group of commands that act on the current document.
        /// </summary>
        public IDocumentICommands Commands { get; private set; }
    }
}
